lib/std/crypto/ripemd160.zig

```diff
@@ -0,0 +1,218 @@
+        .{ 8, 9, 9, 11, 13, 15, 15, 5, 7, 7, 8, 11, 14, 14, 12, 6 },
+        .{ 9, 13, 15, 7, 12, 8, 9, 11, 7, 7, 12, 7, 6, 15, 13, 11 },
+        .{ 9, 7, 15, 11, 8, 6, 6, 14, 12, 13, 5, 14, 13, 13, 7, 5 },
+        .{ 15, 5, 8, 11, 14, 14, 6, 14, 6, 9, 12, 9, 12, 5, 15, 8 },
+        .{ 8, 5, 12, 9, 12, 5, 14, 6, 8, 13, 6, 5, 15, 13, 11, 11 },
+    };
+
+    var left: [5]u32 = undefined;
+    var right: [5]u32 = undefined;
+    std.mem.copy(u32, left[0..], hash[0..5]);
+    std.mem.copy(u32, right[0..], hash[0..5]);
+
+    round(bfn1, -0, 0x00000000, permutations[0], shifts[0], &left, words);
+    round(bfn2, -1, 0x5a827999, permutations[1], shifts[1], &left, words);
+    round(bfn3, -2, 0x6ed9eba1, permutations[2], shifts[2], &left, words);
+    round(bfn4, -3, 0x8f1bbcdc, permutations[3], shifts[3], &left, words);
+    round(bfn5, -4, 0xa953fd4e, permutations[4], shifts[4], &left, words);
+
+    round(bfn5, -0, 0x50a28be6, permutations[5], shifts[5], &right, words);
+    round(bfn4, -1, 0x5c4dd124, permutations[6], shifts[6], &right, words);
+    round(bfn3, -2, 0x6d703ef3, permutations[7], shifts[7], &right, words);
+    round(bfn2, -3, 0x7a6d76e9, permutations[8], shifts[8], &right, words);
+    round(bfn1, -4, 0x00000000, permutations[9], shifts[9], &right, words);
+
+    right[3] = blk: {
+        var x: u32 = right[3];
+        _ = @addWithOverflow(u32, x, left[2], &x);
+        _ = @addWithOverflow(u32, x, hash[1], &x);
+        break :blk x;
+    };
+
+    hash[1] = blk: {
+        var x: u32 = hash[2];
+        _ = @addWithOverflow(u32, x, left[3], &x);
+        _ = @addWithOverflow(u32, x, right[4], &x);
+        break :blk x;
+    };
+
+    hash[2] = blk: {
+        var x: u32 = hash[3];
+        _ = @addWithOverflow(u32, x, left[4], &x);
+        _ = @addWithOverflow(u32, x, right[0], &x);
+        break :blk x;
+    };
+
+    hash[3] = blk: {
+        var x: u32 = hash[4];
+        _ = @addWithOverflow(u32, x, left[0], &x);
+        _ = @addWithOverflow(u32, x, right[1], &x);
+        break :blk x;
+    };
+
+    hash[4] = blk: {
+        var x: u32 = hash[0];
+        _ = @addWithOverflow(u32, x, left[1], &x);
+        _ = @addWithOverflow(u32, x, right[2], &x);
+        break :blk x;
+    };
+
+    hash[0] = right[3];
+}
+
+fn round(
+    comptime bfn: fn (u32, u32, u32) u32,
+    comptime off: i32,
+    comptime constant: u32,
+    comptime p: [16]u5,
+    comptime shifts: [16]u5,
+    hash: *[5]u32,
+    words: [16]u32,
+) void {
+    comptime var r = off;
+    comptime var i: usize = 0;
+    inline while (i < 16) : (i += 1) {
+        hash[@mod(r + 0, 5)] = blk: {
+            var x: u32 = hash[@mod(r + 0, 5)];
+            _ = @addWithOverflow(u32, x, bfn(hash[@mod(r + 1, 5)], hash[@mod(r + 2, 5)], hash[@mod(r + 3, 5)]), &x);
+            _ = @addWithOverflow(u32, x, words[p[i]], &x);
+            _ = @addWithOverflow(u32, x, constant, &x);
+            break :blk x;
+        };
+
+        hash[@mod(r + 0, 5)] = blk: {
```

Comment:
No need for `@mod`, you can use `%`

Reply:
@mod instead of the % operator because the lhs is negative and the rhs is comptime - using % instead here yields undefined and is a compile error